dvc/api.py

```diff
@@ -97,15 +100,17 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
 @contextmanager
 def _make_repo(repo_url, rev=None):
     if not repo_url or urlparse(repo_url).scheme == "":
-        assert rev is None, "Custom revision is not supported for local repo"
+        assert (
+            rev is None
+        ), "Git revisions are not supported for local DVC projects."
```

Comment:
Why not "local repositories"? Not sure I understand what's a _local DVC project_.

Reply:
A DVC project isn't always a repo. It can be initialized with `--no-scm` and you can provide the local path to that project in the file system to the `repo` argument (misleading name BTW) of API functions.